src/gt4py/next/iterator/embedded.py

```diff
@@ -1392,7 +1391,12 @@ def _closure_runner():
                         col_pos[column.axis] = k
                         assert _is_concrete_position(col_pos)
                         ordered_indices = get_ordered_indices(out.axes, col_pos)
-                        out.field_setitem(ordered_indices, res[k])
+                        if isinstance(res, tuple):
+                            out.field_setitem(
+                                ordered_indices, tuple(res[i][k] for i in range(len(res)))
+                            )  # TODO(tehrengruber): only works for scalars
+                        else:
+                            out.field_setitem(ordered_indices, res[k])
```

Comment:
is this fixed in https://github.com/GridTools/gt4py/pull/1141/files? or is it another instance of that problem?

Reply:
Not sure actually, might be two different things. This is not clean however as it only works for tuples not nested tuples.